### ISSUE TITLE
Blazor WASM hotreload with StrawberryShake code generation

### ISSUE THREAD

**Issue by sourcejon** [ChilliCream/hotchocolate#4008]
**Describe the bug**
Hotreload fails on a blazor wasm project when using StrawberryShake code generation

**To Reproduce**
1. create a blazor wasm project using dotnet 6.0.100-preview.6.21355.2
2. add "hotReloadProfile": "blazorwasm" to your launch profile in launchSettings.json
3. run project using dotnet watch, check output for the following message: watch : Hot reload enabled. For a list of supported edits, see https://aka.ms/dotnet/hot-reload. Press "Ctrl + R" to restart.
4. make a simple edit to index.razor and ensure hotreload is working
5. add StrawberryShake tooling for code generation and configure as per https://chillicream.com/docs/strawberryshake/get-started
6. run dotnet watch and make a simple change as per step 4.

**Expected behavior**
hotreload continues to work and code generators does not fire

**Desktop (please complete the following information):**
Win 10, Chrome 91

**Additional context**
watch : StrawberryShake.CodeGeneration.CSharp.Analyzers\StrawberryShake.CodeGeneration.CSharp.Analyzers.CSharpClientGenerator\SourceGraphQLClient.StrawberryShake.cs(1332,57): error CS0229: Ambiguity between 'GetBrokerByIDBuilder._idSerializer' and 'GetBrokerByIDBuilder._idSerializer'
watch : StrawberryShake.CodeGeneration.CSharp.Analyzers\StrawberryShake.CodeGeneration.CSharp.Analyzers.CSharpClientGenerator\SourceGraphQLClient.StrawberryShake.cs(1338,102): error CS0121: The call is ambiguous between the following methods or properties: 'GetBrokerByIDBuilder.DeserializeNonNullableInt32(JsonElement?)' and 'GetBrokerByIDBuilder.DeserializeNonNullableInt32(JsonElement?)'

**Comment by psalmeight** [comment]
@sourcejon were you able to run strawberryshake on .net  6 prev 6? did you had any issues related with https://github.com/ChilliCream/hotchocolate/issues/3959?

**Comment by sourcejon** [comment]
@psalmeight No issues with #3959 , it all run's perfect except for hotreload support. It's worth noting I do have .net 5 installed on my system.


`PS C:\Users\X\source\repos\X\src\WebApps\BlazorHotReload> dotnet graphql init https://workshop.chillicream.cloud/graphql/ -n ConferenceClient -p .\BlazorHotReload\
Download schema started.
Download schema completed in 727 ms
Client configuration started.
Client configuration completed in 167 ms
PS C:\Users\X\source\repos\X\src\WebApps\BlazorHotReload> dotnet --version
6.0.100-preview.6.21355.2`

**Comment by dmg-hamann** [comment]
I am having the same issue. For the time beeing you can set "emitGeneratedCode"in .graphqlrc.json to false. Without the generated code in your solution dotnet watch works fine. But of course you will lack intelli sense.

**Comment by michaelstaib** [comment]
Hey there, we will address this soon ... We have just switched to the preview 7 SDK and will slowly upgrade project targets.

**Comment by MisinformedDNA** [comment]
Where are we with this?

**Comment by jorrit** [comment]
I have something similar. I use a generated class as Model in my cshtml, which is dynamically compiled. When displaying the view, I get:

```
InvalidOperationException: The model item passed into the ViewDataDictionary is of type 'SomeClass', but this ViewDataDictionary instance requires a model item of type 'SomeClass'.
```

I also have many build warnings along the form:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS0436	The type 'SomeClass' in 'StrawberryShake.CodeGeneration.CSharp.Analyzers\StrawberryShake.CodeGeneration.CSharp.Analyzers.CSharpClientGenerator\MyClient.StrawberryShake.cs' conflicts with the imported type 'SomeClass' in Project, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'StrawberryShake.CodeGeneration.CSharp.Analyzers\StrawberryShake.CodeGeneration.CSharp.Analyzers.CSharpClientGenerator\MyClient.StrawberryShake.cs'.	MyProject	C:\Project\obj\Debug\net5.0\Razor\Views\View.cshtml.g.cs	45	Active
```

**Comment by michaelstaib** [comment]
This is now fixed with 12.7
